Swerve/SwerveModules/SwerveModule.java

```diff
@@ -25,15 +26,36 @@
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import org.littletonrobotics.junction.AutoLog;
+import org.littletonrobotics.junction.Logger;
 
 public abstract class SwerveModule {
     public SwerveModuleConstants swerveModuleConstants;
     protected CANcoder angleEncoder;
     protected Rotation2d lastAngle;
-    
+
+    private final ModuleIOInputsAutoLogged inputs = new ModuleIOInputsAutoLogged();
```

Comment:
Where exactly is this class created (just comment below here please)

Reply:
AKit made the @Autolog notation which, when put on ModuleIOInputs, creates another class with the same path with fromLog and toLog methods that automatically log the class variables periodically. AKit uses this annotation in their template swerve as well as the Logger.processInputs and not sure why. Docs are here https://docs.advantagekit.org/data-flow/recording-outputs/annotation-logging

Reply:
oh so akit created a dynamically generated class?

so if I was to do @autolog on a class called VishruthRaoInputs it would generate a called VishruthRaoInputsAutoLogged?

pretty interesting

consider this resolved